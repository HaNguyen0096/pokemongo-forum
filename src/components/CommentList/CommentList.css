.commentList{
  display: block;
  padding: 0;
  margin-left: 10px;
  margin-right: 10px;
}

.threadBar{

  display: inline-block;
  position: relative;
  font-weight: bold;
  vertical-align: bottom;
  margin-bottom: 10px;
}

.threadTitle{
  text-decoration: none;
}


.threadContent{
  margin-bottom: 10px;
}

.commentTime{
  color: rgb(113,113,113);
  margin-bottom: 10px;
}

.cmthr{
  width: 100%
}

.dltComment{
  margin-bottom: 10px;
}








