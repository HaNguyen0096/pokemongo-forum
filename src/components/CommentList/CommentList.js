import React, { Component} from 'react'

export default class CommentList extends Component{
  render() {
    const { comment } = this.props
    console.log(comment)
    return (
      <div className='ThreadList'>
        <h3>{comment.content}</h3>
      </div>      
    )
  }
}