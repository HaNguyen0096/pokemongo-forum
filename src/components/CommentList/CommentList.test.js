import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import CommentList from './CommentList'

const testComment = [
  {
    id: 1,
    content: 'test 1',
    user_id: 1,
    thread_id: 1,
    modified: '2020-02-22T21:28:32.615Z'
  },
]



describe('CommentList Component', () => {

  it('renders CommentList without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                      <CommentList comment={testComment}/>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // it('renders the UI for CommentList as expected', () => {
  //   const tree = renderer
  //     .create(<BrowserRouter>
  //               <CommentList comment={testComment} />
  //             </BrowserRouter>)
  //     .toJSON();
  //   expect(tree).toMatchSnapshot();  
  // });

  // it('renders CommentList by default', () => {
  //   const wrapper = shallow(<CommentList comment={testComment} />)
  //   expect(toJson(wrapper)).toMatchSnapshot()
  // });

})


