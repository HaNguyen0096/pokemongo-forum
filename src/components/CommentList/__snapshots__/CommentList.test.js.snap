// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentList Component renders CommentList by default 1`] = `
<div
  className="commentList"
>
  <div
    className="commentContent"
  >
    <h5 />
  </div>
  <div
    className="commentTime"
  >
    March 24th 2020, 7:07:10 pm
  </div>
  <div
    className="dltComment"
  >
    <DeleteThread
      comment={
        Array [
          Object {
            "content": "test 1",
            "id": 1,
            "modified": "2020-02-22T21:28:32.615Z",
            "thread_id": 1,
            "user_id": 1,
          },
        ]
      }
      onDeleteThread={[Function]}
    />
  </div>
  <hr
    className="cmthr"
  />
</div>
`;

exports[`CommentList Component renders the UI for CommentList as expected 1`] = `
<div
  className="commentList"
>
  <div
    className="commentContent"
  >
    <h5 />
  </div>
  <div
    className="commentTime"
  >
    March 24th 2020, 7:07:10 pm
  </div>
  <div
    className="dltComment"
  >
    <div
      className="Delete_Comment"
    >
      <button
        className="Delete_Button"
        onClick={[Function]}
        type="button"
      >
         
        remove
      </button>
    </div>
  </div>
  <hr
    className="cmthr"
  />
</div>
`;

exports[`CommentListmponent renders CommentList default 1`] = `
<div
  className="commentList"
>
  <div
    className="commentContent"
  >
    <h5 />
  </div>
  <div
    className="commentTime"
  >
    March 24th 2020, 7:05:46 pm
  </div>
  <div
    className="dltComment"
  >
    <DeleteThread
      comment={
        Array [
          Object {
            "content": "test 1",
            "id": 1,
            "modified": "2020-02-22T21:28:32.615Z",
            "thread_id": 1,
            "user_id": 1,
          },
        ]
      }
      onDeleteThread={[Function]}
    />
  </div>
  <hr
    className="cmthr"
  />
</div>
`;

exports[`CommentListmponent renders the UI for CommentList expected 1`] = `
<div
  className="commentList"
>
  <div
    className="commentContent"
  >
    <h5 />
  </div>
  <div
    className="commentTime"
  >
    March 24th 2020, 7:05:46 pm
  </div>
  <div
    className="dltComment"
  >
    <div
      className="Delete_Comment"
    >
      <button
        className="Delete_Button"
        onClick={[Function]}
        type="button"
      >
         
        remove
      </button>
    </div>
  </div>
  <hr
    className="cmthr"
  />
</div>
`;
